include/mbedtls/platform_util.h

```diff
@@ -129,6 +129,13 @@ MBEDTLS_DEPRECATED typedef int mbedtls_deprecated_numeric_constant_t;
  */
 #define MBEDTLS_CHECK_RETURN_OPTIONAL
 
+/** \def MBEDTLS_IGNORE_RETURN
+ *
+ * Silences warning about unused return value given by functions
+ * with \c MBEDTLS_CHECK_RETURN attribute.
+ */
+#define MBEDTLS_IGNORE_RETURN(result) if( result ) {}
```

Comment:
Might we want to do this inside a `do { ... }while( 0 )` -style idiom so that something like:
```C
MBEDTLS_IGNORE_RETURN(my_function());
```
Doesn't syntax error?

Reply:
The `if(…){}` implementation doesn't lead to a syntax error. It also leads to the correct parsing with `if (condition) MBEDTLS_IGNORE_RETURN(…); else something_else();`, which is the point of the `do … while (0)` idiom. I do agree that using the do-while idiom would be better: it would be less surprising, and also less error-prone since `if (condition) MBEDTLS_IGNORE_RETURN(…) else something_else();` (missing semicolon before `else`) would have a surprising parsing.

Investigating a bit more, it seems that while `(void) (result)` doesn't work to ignore `warn_unused_result` in GCC, `(void) !(result)` does work (https://stackoverflow.com/questions/40576003/ignoring-warning-wunused-result, https://gcc.gnu.org/bugzilla/show_bug.cgi?id=66425#c34, https://gcc.gnu.org/bugzilla/show_bug.cgi?id=25509). So I propose to change to `(void) !(result)`.

I also propose to not redefine `MBEDTLS_IGNORE_RETURN` if it's already defined. Should it be an official configuration interface, listed in `config.`?

Reply:
Hmm
I did some local tests of your first `if-else` example and it seems to fail parsing with `error: ‘else’ without a previous ‘if’`. Why this happens is not entirely clear to me.

Reply:
Ah, yes, my bad, my mental parser misparsed this. Indeed it works with something like `if (result) (void)0; else 0` but not with braces.

In any case I'm going to change the expansion to be an expression.